### PR TITLE
doc: net: Add HTTP server documentation 

### PR DESCRIPTION
**Pull request by rlubos**
* Add documentation page for HTTP server functionality.
* Rename existing HTTP documentation to HTTP client, as it only covers the
client library.
* Add missing doxygen groups for HTTP service and server APIs.
* Improve doxygen API documentation when missing.

### CODE REVIEW

**Review by jukkar — COMMENTED** [review]
Did a cursory review, looks good. I will revisit after the websocket stuff is merged.

**Comment by rlubos** [comment]
Rebased, resovled conflicts and added a short section about websocket resources. Good to go from my point of view.

**Review by kartben — COMMENTED** [review]
I'm off today, but a small nit in the meantime:)
Looks really good after a quick look, thanks for this!

File: doc/connectivity/networking/api/http_server.rst [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+###########
+
+.. contents::
+    :local:
+    :depth: 2
+
+Overview
+********
+
+Zephyr provides a HTTP server library, which allows to register HTTP services
```
[the remark this answers is not in the document]
Comment (rlubos) [reply doc/connectivity/networking/api/http_server.rst]: Fixed

**Review by pdgendt — APPROVED** [review]

**Review by kartben — APPROVED** [review]
Brilliant, thanks!

**Review by jukkar — APPROVED** [review]

**Pull request merged**